jsettlers.graphics/src/jsettlers/graphics/utils/UIList.java

```diff
@@ -49,19 +49,18 @@
 		UIListItem getItem(T item);
 	}
 
-	public UIList(List<? extends T> items, ListItemGenerator<T> generator,
-			float itemheight) {
-		this.items = items;
+	public UIList(List<? extends T> items, ListItemGenerator<T> generator, float itemheight) {
+		setItems(items);
 		this.generator = generator;
 		this.itemheight = itemheight;
-		if (items.size() > 0) {
-			activeItem = items.get(0);
-		}
 	}
 
 	public void setItems(List<? extends T> list) {
 		synchronized (itemsMutex) {
 			this.items = list;
+			if (items.size() > 0) {
```

Comment:
This is not the way it was intended. Lists are often updated by adding items. If we reset the active item on every list update, we reset the user selection.